sentry/src/main/java/io/sentry/SentryClient.java

```diff
@@ -521,6 +490,37 @@ public void captureSession(final @NotNull Session session, final @Nullable Objec
     return event;
   }
 
+  private <T extends SentryBaseEvent> T applyScope(
+      final @NotNull T sentryBaseEvent, final @Nullable Scope scope) {
```

Comment:
there's already a NPE check to call this method, hence `Scope` can be non null

Reply:
For the transaction we use this method directly, that's why it's kept as nullable.